compiler/compile/OMRCompilation.cpp

```diff
@@ -702,12 +705,25 @@ bool OMR::Compilation::isPotentialOSRPointWithSupport(TR::TreeTop *tt)
 int32_t
 OMR::Compilation::getOSRInductionOffset(TR::Node *node)
    {
-   if (self()->getHCRMode() == TR::osr && node->getOpCodeValue() != TR::asynccheck)
-      return 3;
+   if (self()->getHCRMode() == TR::osr)
+      {
+      switch (node->getOpCodeValue())
+         {
+         case TR::monent: return 1;
+         case TR::asynccheck: return 0;
+         default: return 3;
+         }
+      }
    return 0;
    }
 
 bool
+OMR::Compilation::requiresPreOSRPoint(TR::Node *node)
+   {
+   return node->getOpCodeValue() != TR::monent;
```

Comment:
Could you please add a comment here explaining why monent is treated differently to async check and call in terms of a PreOSRPoint ?